src/dialogs/area-filter/area-filter-dialog.ts

```diff
@@ -0,0 +1,208 @@
+    await this.updateComplete;
+    this._createSortable();
+  }
+
+  public closeDialog(): void {
+    this._dialogParams = undefined;
+    this._hidden = [];
+    this._areas = [];
+    this._destroySortable();
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  private _submit(): void {
+    const order = this._areas.filter((area) => !this._hidden.includes(area));
+    const value: AreaFilterValue = {
+      hidden: this._hidden,
+      order,
+    };
+    this._dialogParams?.submit?.(value);
+    this.closeDialog();
+  }
+
+  private _cancel(): void {
+    this._dialogParams?.cancel?.();
+    this.closeDialog();
+  }
+
+  private async _createSortable() {
+    const Sortable = (await import("../../resources/sortable")).default;
+    if (this._sortable) return;
+    this._sortable = new Sortable(this.shadowRoot!.querySelector(".areas")!, {
+      animation: 150,
+      fallbackClass: "sortable-fallback",
+      onChoose: (evt: SortableEvent) => {
+        (evt.item as any).placeholder =
+          document.createComment("sort-placeholder");
+        evt.item.after((evt.item as any).placeholder);
+      },
+      onEnd: (evt: SortableEvent) => {
+        // put back in original location
+        if ((evt.item as any).placeholder) {
+          (evt.item as any).placeholder.replaceWith(evt.item);
+          delete (evt.item as any).placeholder;
+        }
+        this._dragged(evt);
+      },
+    });
+  }
+
+  private _destroySortable() {
+    this._sortable?.destroy();
+    this._sortable = undefined;
+  }
+
+  private _dragged(ev: SortableEvent): void {
+    if (ev.oldIndex === ev.newIndex) return;
+
+    const areas = this._areas.concat();
+
+    const option = areas.splice(ev.oldIndex!, 1)[0];
+    areas.splice(ev.newIndex!, 0, option);
+
+    this._areas = areas;
+  }
+
+  protected render() {
+    if (!this._dialogParams || !this.hass) {
+      return nothing;
+    }
+
+    const allAreas = this._areas;
+
+    return html`
+      <ha-dialog
+        open
+        @closed=${this._cancel}
+        .heading=${this._dialogParams.title ?? "Areas"}
+      >
+        <mwc-list class="areas" @action=${this._action}>
+          ${repeat(
+            allAreas,
+            (area) => area,
+            (area, _idx) => {
+              const isVisible = !this._hidden.includes(area);
+              return html`
+                <ha-list-item
+                  class=${classMap({ hidden: !isVisible })}
+                  hasMeta
+                  graphic="icon"
+                >
+                  <ha-svg-icon
+                    class="handle"
+                    .path=${mdiDrag}
+                    slot="graphic"
+                  ></ha-svg-icon>
+                  ${this.hass!.areas[area]?.name || area}
+                  <ha-svg-icon
+                    class="action"
+                    .path=${isVisible ? mdiEye : mdiEyeOff}
+                    slot="meta"
+                  ></ha-svg-icon>
```

Comment:
I expected this to be a button, and not the entire row to be clickable?

Reply:
Yes, we can. Let's discuss that during front-end meeting.